src/Voron/Data/Sets/Set.cs

```diff
@@ -518,8 +518,11 @@ public Iterator(Set parent)
             }
 
             public bool? MaybeSeek(long from)
-            {
-                if (_it.IsInRange(from))
+            {                
+                // TODO: In case that we are in the right location but we have passed over,
+                //       we may be able to use a optimized Seek method instead which would
+                //       avoid getting the page and just start over in the current segment.
+                if (Current < from && _it.IsInRange(from)) 
```

Comment:
Pretty sure that we can just reset the iterator in this case.
IIRC, just re-creating the iterator should be cheap enough.

*However*, need to make sure that the `from` falls within the current page, remember that it may actually be a value from a _previous_ page, no?

Reply:
Yes. That was my though. If it is still in the current page we may have that option available to us.

Reply:
OK. Update on this... not so easy... If you try to avoid the `FindPageFor` it may happen that you already moved on the leaf page, but we need to find a way to query the first element efficiently. I will leave it as TODO, you can give it a try. Want to have this mergeable ASAP.